### PR TITLE
implement certified token proposal

### PR DESCRIPTION
**Pull request by ylsGit**
- Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/okex/okchain/blob/develop/CONTRIBUTING.md#pr-targeting))

- [ ] Linked to github-issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Wrote tests
- [ ] Updated relevant documentation (`docs/`)
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Reviewed `Files changed` in the github PR explorer

______

For Admin Use:
- Added appropriate labels to PR (ex. wip, ready-for-review, docs)
- Reviewers Assigned
- Squashed all commits, uses message "Merge PR #XYZ: [title]" ([coding standards](https://github.com/okex/coding/blob/master/README.md#merging-a-pr))

### CODE REVIEW

File: x/token/handler.go [hunk 1]
```diff
@@ -393,3 +399,72 @@ func handleMsgTokenModify(ctx sdk.Context, keeper Keeper, msg types.MsgTokenModi
 	)
 	return sdk.Result{Events: ctx.EventManager().Events()}
 }
+
+func handleMsgTokenActive(ctx sdk.Context, keeper Keeper, msg types.MsgTokenActive, logger log.Logger) sdk.Result {
+	certifiedToken := keeper.GetCertifiedToken(ctx, msg.ProposalID)
```
Comment (xiangjianmeng) [inline x/token/handler.go:404]: The token info is stored in proposal, so I think handleMsgTokenActive can get info from proposal by proposal ID to activate the certified token.

File: x/token/client/cli/tx.go [hunk 2]
```diff
@@ -477,3 +483,101 @@ func getCmdTokenEdit(cdc *codec.Codec) *cobra.Command {
...
+     "description": "Bitcoin in testnet，1:1 anchoring with Bitcoin",
+     "symbol": "tbtc",
+     "whole_name": "Testnet Bitcoin",
+     "total_supply": "21000000",
+     "owner": "okchain170ydluqy9qnza6t7mt5rqtqw5p3esdrz00hvm2",
+     "mintable": false
+   },
+  "deposit": [
+    {
+      "denom": okt,
```
[the remark this answers is not in the document]
Comment (ylsGit) [reply x/token/client/cli/tx.go:515]: replaced

File: x/token/handler.go (hunk 1 above)
Comment (ylsGit) [reply x/token/handler.go]: Still keep this way, but add delete operation
When the certified token is activated, delete the certified token data

File: x/token/types/params.go [hunk 3]
```diff
@@ -14,26 +15,33 @@ const (
...
 var _ params.ParamSet = &Params{}
 
 // mint parameters
 type Params struct {
-	FeeIssue     sdk.DecCoin `json:"issue_fee"`
-	FeeMint      sdk.DecCoin `json:"mint_fee"`
-	FeeBurn      sdk.DecCoin `json:"burn_fee"`
-	FeeModify    sdk.DecCoin `json:"modify_fee"`
-	FeeChown     sdk.DecCoin `json:"transfer_ownership_fee"`
+	FeeIssue  sdk.DecCoin `json:"issue_fee"`
```
Comment (summerpro) [inline x/token/types/params.go:36]: the json contents are not equal to the name of param.
it is best to write like this:
```
FeeIssue  sdk.DecCoin `json:"fee_issue"`
```

File: x/token/types/proposal.go [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+			fmt.Sprintf("failed to submit CertifiedToken proposal because description is longer than max length of %d", govtypes.MaxDescriptionLength))
+	}
+
+	if ctp.ProposalType() != ProposalTypeCertifiedToken {
+		return govtypes.ErrInvalidProposalType(DefaultCodespace,
+			ctp.ProposalType())
+	}
+
+	// check owner
+	if ctp.Token.Owner.Empty() {
```
Comment (summerpro) [inline x/token/types/proposal.go:100]: it is best to validate the format of the owner

**Comment by zhongqiuwood** [comment]
This feature is suspended.

**Pull request closed without merging**